portablemc/optifine.py

```diff
@@ -0,0 +1,659 @@
+                    "--tweakClass" in version.metadata["arguments"]["game"] and
+                    "optifine.OptiFineTweaker" in version.metadata["arguments"]["game"]):
+                    return launchwrapperseemscorrect and oflibseemscorrect
+                else:
+                    return False
+        return False
+    def _finalize_optifine_internal(self, watcher: Watcher) -> None:
+        version = VersionHandle(self.version, self.context.versions_dir / self.version)
+
+        if not self.check_of_install(version): # if the version is not installed, install it
+            watcher.handle(OptifineStartInstallEvent())
+            jar_path=self.context.versions_dir / self.version / "ofInstaller.jar"
+            with zipfile.ZipFile(jar_path, "r") as jar:
+                version=VersionHandle(self.version, self.context.versions_dir / self.version)
+                try:
+                    launchwrapper = f"optifine:launchwrapper-of:{jar.open('launchwrapper-of.txt').read().decode('utf-8').strip()}"
+                    launchwrapper_version=jar.open("launchwrapper-of.txt").read().decode("utf-8").strip()
+                except KeyError:
+                    launchwrapper = "net.minecraft:launchwrapper:1.12"
+                minecraft_ver = VersionHandle(self.mcver(), self.context.versions_dir / self.mcver())
+                if minecraft_ver.read_metadata_file():
+                    parent_data = minecraft_ver.metadata
+                else:
+                    mcver_url=self.manifest.get_version(self.mcver)["url"]
+                    res = http_request("GET", mcver_url, accept="application/json")
+                    parent_data = res.json()
+                    minecraft_ver.metadata = parent_data
+                    minecraft_ver.write_metadata_file()
+
+                # patch minecraft version jar to build Optifine library
+                of_lib_dir=self.context.libraries_dir/"optifine"/"OptiFine"/f"{self.mcver()}_{self.loader()}"
+                if not of_lib_dir.exists(): # makes the library directory
+                    of_lib_dir.mkdir(parents=True,exist_ok=True)
+                Patcher.process(self.context.versions_dir / self.version / f"{self.version}.jar",
+                                self.context.versions_dir / self.version / "ofInstaller.jar",
+                                self.context.libraries_dir/"optifine"/"OptiFine"/f"{self.mcver()}_{self.loader()}"/f"OptiFine-{self.mcver()}_{self.loader()}.jar",
+                                watcher=watcher)
+
+                launchwrapper_sha1=None
+                if not launchwrapper=="net.minecraft:launchwrapper:1.12":
+                    # install launchwrapper library if it is not default launchwrapper
+                    file_name = f"launchwrapper-of-{launchwrapper_version}.jar"
+                    dir_dest = self.context.libraries_dir / f"optifine/launchwrapper-of/{launchwrapper_version}"
```

Comment:
Use the separator pattern: `self.context.libraries_dir / "optifine" / "launchwrapper-of" / launchwrapper_version`